packages/x-date-pickers/src/internals/models/props/baseToolbarProps.ts

```diff
@@ -1,23 +1,33 @@
 import * as React from 'react';
 import { CalendarOrClockPickerView } from '../views';
-import type { PickerOnChangeFn } from '../../hooks/useViews';
-import type { ExportedDateCalendarProps } from '../../../DateCalendar/DateCalendar';
-import type { ExportedClockPickerProps } from '../../../ClockPicker/ClockPicker';
-import { PickerStatePickerProps } from '../../hooks/usePickerState';
 
-export interface BaseToolbarProps<TDate, TValue>
-  extends ExportedDateCalendarProps<TDate>,
-    ExportedClockPickerProps<TDate>,
-    Omit<PickerStatePickerProps<TValue>, 'onDateChange'> {
-  ampmInClock?: boolean;
-  dateRangeIcon?: React.ReactNode;
+export interface BaseToolbarProps<TValue> extends ExportedBaseToolbarProps {
   isLandscape: boolean;
-  onChange: PickerOnChangeFn<TDate>;
-  openView: CalendarOrClockPickerView;
-  setOpenView: (view: CalendarOrClockPickerView) => void;
-  timeIcon?: React.ReactNode;
+  onChange: (newValue: TValue) => void;
+  value: TValue;
+  view: CalendarOrClockPickerView;
+  onViewChange: (view: CalendarOrClockPickerView) => void;
+  views: readonly CalendarOrClockPickerView[];
+  disabled?: boolean;
+  readOnly?: boolean;
+  // TODO v6: Remove
+  isMobileKeyboardViewOpen: boolean;
+  // TODO v6: Remove
+  toggleMobileKeyboardView: () => void;
+}
+
+export interface ExportedBaseToolbarProps {
+  /**
+   * Date format.
+   */
   toolbarFormat?: string;
+  /**
+   * Mobile picker date value placeholder, it is displayed when `value` === `null`.
+   * @default '–'
+   */
   toolbarPlaceholder?: React.ReactNode;
+  /**
+   * Mobile picker title.
```

Comment:
For me we can add it to the locale text indeed.

Reply:
I've done some tinkering and am not sure—was it a community requirement or our decision to put `label` in the `toolbarTitle` if it is defined? 🤔
Honestly—I do not find it particularly useful as probably most use-cases include a `label` defined—then `localeText.<pickerComponent>DefaultToolbarTitle` field to define a different `toolbarTitle` sort of looses a point 🙈 

The current priority seems the following:
`props.toolbarTitle ?? props.label ?? localeText.<pickerComponent>DefaultToolbarTitle`

I'm not saying it is wrong—just that the last case is probably used too rarely.

Reply:
So you would go for the following prioriy ?

`localeText.<pickerComponent>ToolbarTitle` and drop all the existing ?

Reply:
What is @alexfauquette take on this one?

Reply:
> But if we were to drop toolbarTitle—I'd say we should deprecate it in v5 first. thinking

We are moving it in this PR so I think if we want to drop it, we should do it now.

For v5 what would you do to deprecate ?
If we deprecate `props.toolbarTitle`, should we create the new locale text and have `localeText.<pickerComponent>ToolbarTitle ?? localeText.<pickerComponent>DefaultToolbarTitle` ?

Reply:
> For v5 what would you do to deprecate ?

Yeah—it's a tricky prop to deprecate because of the existing application priority 🙈 
Maybe it wouldn't be too big of trouble to just drop it in v6, because of the low importance of this prop. 🤔

Reply:
I do agree with the priority you propose
If @alexfauquette if fine with it, I can add it to this PR

Reply:
Not sure I got all the proposal.

You suggest to remove text provided by props, and replace them by the following:

```ts
localeText.<pickerComponent>ToolbarTitle ?? localeText.<pickerComponent>DefaultToolbarTitle
```

Maybe better to us

```ts
localeText.toolbarTitle ?? localeText.<pickerComponent>DefaultToolbarTitle
```

with the `<pickerComponent>DefaultToolbarTitle` being provided in community translations, and apply evry where. It can be overridden by dev if they want an customization at the scale of the app. 

And `toolbarTitle` should be used for component specific customization (replace the prop purpose) In such case we don't need to have a key per component

```diff
<DatePicker
	localeText={{
-			DatePickerToolbarTitle: "Meeting date",
+			toolbarTitle: "Meeting date",
	}}
/>
```

Reply:
> You suggest to remove text provided by props, and replace them by the following:

No, just by the following:

```
localeText.<pickerComponent>ToolbarTitle ?? localeText.<pickerComponent>DefaultToolbarTitle
```

Now that we have the `localeText` prop, people will be able to override with it, making the `toolbarTitle` almost the only text prop not in the `localeText`

```diff
<DatePicker
-  toolbarTitle="MeetingDate"
+  localeText={{ datePickerToolbarTitle: 'Meeting date' }}
/>
```

Reply:
Ok, so I confirm, this modification sound weird:
```diff
<DatePicker
-  toolbarTitle="MeetingDate"
+  localeText={{ datePickerToolbarTitle: 'Meeting date' }}
/>
```

I prefer this one to avoid repeating you are considering a date picker

```
<DatePicker
-  toolbarTitle="MeetingDate"
+  localeText={{ toolbarTitle: 'Meeting date' }}
/>
```

Reply:
But how do we put a default value that depends on the picker ? Like we have today

Reply:
I taught it was the role of `localeText.<pickerComponent>DefaultToolbarTitle`

Reply:
OK so you would keep a `localeText.toolbarTitle` with no default value in the locale files ?
It is something we currently never have but that we could introduce.

Reply:
Yes

Reply:
@LukasTy what are your thoughts on this ?

Reply:
I would agree with Alex, it definitely would seem strange to provide component prefixed `toolbarTitle` prop via `localeText` when passing it to the actual component.

Reply:
OK this is a huge pain to has a different typing between the localization provider used by the user and the one accessed inside the component after the picker pre-processing.

I propose to keep the `toolbarTitle` prop on the toolbar for now :grimacing:

Reply:
https://github.com/mui/mui-x/pull/6445/commits/42fd23776574f243dcb74f7c90a57c6769f8ee99

Here it is